tools/nxstyle.c

```diff
@@ -831,6 +837,51 @@ int main(int argc, char **argv, char **envp)
                ERROR("Missing file header comment block", lineno, 1);
             }
 
+          if (lineno == 2)
+            {
+              if (line[n] == '*' && line[n + 1] == '\n')
+                {
+                  ERROR("Missing relative file path in file header", lineno,
+                        n);
+                }
+              else if (isspace(line[n + 2]))
+                {
+                  ERROR("Too many whitespaces before relative file path",
+                        lineno, n);
+                }
+#ifdef TOPDIR
+              else
+                {
+                  /* TOPDIR macro should have been defined via Makefile and
+                   * it is required for evaluating the relative file path in
+                   * the file header.
+                   * Otherwise, skip this verification.
+                   */
+
+                  /* Calculate the relative file path length minus the '\n'
+                   * character at the end of the string.
+                   */
+
+                  size_t filepath_len = strlen(&line[n + 2]) - 1;
+
+                  char *basedir = strstr(g_file_name, TOPDIR);
+                  if (basedir != NULL)
+                    {
+                      /* Add 1 to the offset for the slash character */
+
+                      size_t offset = strlen(TOPDIR) + 1;
+
+                      if (strncmp(&line[n + 2], basedir + offset,
+                                  filepath_len) != 0)
```

Comment:
can't catch the error like this:
```
file name: ~/nuttx/sched/clock/Make.defs
comment: sched/clock/Make.def
```
It's better to remove '\n' from line and use strcmp instead

Reply:
You're right.
But instead of modifying `line` I'll duplicate it to avoid messing with further verifications.